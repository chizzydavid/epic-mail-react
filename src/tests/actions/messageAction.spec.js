/* eslint-disable arrow-parens */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { 
  getMessages, 
  setActiveMsgTab, 
  setMessages, 
  messageError, 
  messageFeedback 
} from '../../store/actions/messageActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('authAction', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('should retrieve messages of different categories', done => {
    nock('https://localhost:3000')
      .get('/api/messgages')
      .reply(200, []);

    return store.dispatch(getMessages('all')).then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  it('should set active message tab', () => {
    store.dispatch(setActiveMsgTab('sent'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should set messages', () => {
    store.dispatch(setMessages('drafts', []));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should set messages error', () => {
    store.dispatch(messageError({}));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should set messages feedback', () => {
    store.dispatch(messageFeedback({}));
    expect(store.getActions()).toMatchSnapshot();
  });  

});
