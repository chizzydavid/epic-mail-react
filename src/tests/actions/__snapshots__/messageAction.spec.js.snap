// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authAction should retrieve messages of different categories 1`] = `
Array [
  Object {
    "type": "BEGIN_MSG_REQUEST",
  },
  Object {
    "payload": Object {
      "error": "No Authentication Token Provided.",
      "status": 401,
    },
    "type": "MSG_ERROR",
  },
]
`;

exports[`authAction should set active message tab 1`] = `
Array [
  Object {
    "payload": "sent",
    "type": "SET_ACTIVE_TAB",
  },
]
`;

exports[`authAction should set messages 1`] = `
Array [
  Object {
    "payload": Array [],
    "type": "SET_DRAFTS",
  },
]
`;

exports[`authAction should set messages error 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "MSG_ERROR",
  },
]
`;

exports[`authAction should set messages feedback 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "MSG_FEEDBACK",
  },
]
`;
